test/functional/test_framework/test_framework.py

```diff
@@ -666,7 +666,12 @@ def sync_mempools(self, nodes=None, wait=1, timeout=60, flush_scheduler=True):
 
     def sync_all(self, nodes=None):
         self.sync_blocks(nodes)
-        self.sync_mempools(nodes)
+        try:
+            self.sync_mempools(nodes)
+        except JSONRPCException as e:
+            if (e.error["code"]) != -33 or ("Mempool disabled or instance not found" not in e.error['message']):
+                raise e
+            self.log.info("Mempool disabled or instance not found, unable to sync mempools of the nodes")
```

Comment:
I think this check needs to be inside the sync_method. (Imagine a test with multiple nodes, where some have a mempool and other do not)